src/CoreWCF.AzureQueueStorage/src/CoreWCF/Channels/AzureQueueStorageQueueTransport.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO.Pipelines;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+using CoreWCF.Configuration;
+using CoreWCF.Queue;
+using CoreWCF.Queue.Common;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace CoreWCF.Channels
+{
+    internal class AzureQueueStorageQueueTransport : IQueueTransport
+    {
+        private MessageQueue _queueClient;
+        private DeadLetterQueue _deadLetterQueueClient;
+        private TimeSpan _receiveMessagevisibilityTimeout;
+        private ILogger<AzureQueueStorageQueueTransport> _logger;
+        private Uri _baseAddress;
+
+        public AzureQueueStorageQueueTransport(IServiceDispatcher serviceDispatcher, IServiceProvider serviceProvider)
+        {
+            _queueClient = serviceProvider.GetRequiredService<MessageQueue>();
+            _deadLetterQueueClient = serviceProvider.GetRequiredService<DeadLetterQueue>();
+            _logger = serviceProvider.GetRequiredService<ILogger<AzureQueueStorageQueueTransport>>();
+            _baseAddress = serviceDispatcher.BaseAddress;
+            _receiveMessagevisibilityTimeout = TransportDefaults.ReceiveMessagevisibilityTimeout;
+        }
```

Comment:
This should be a property on the transport binding element. I don't think it's necessary to include on the binding as we should have some default and modifying it seems like a more advanced scenario.

Reply:
I am setting the default value in the constructor. but if we want to move it to transport binding element, how do i access that binding element object here?

Reply:
passed the current object and used it to access the variable